conda_forge_tick/migrators/cstdlib.py

```diff
@@ -0,0 +1,236 @@
+import os
+import re
+
+from conda_forge_tick.migrators.core import MiniMigrator
+from conda_forge_tick.migrators.libboost import _replacer, _slice_into_output_sections
+
+pat_stub = re.compile(r"(c|cxx|fortran)_compiler_stub")
+rgx_idt = r"(?P<indent>\s*)-\s*"
+rgx_pre = r"(?P<compiler>\{\{\s*compiler\([\"\']"
+rgx_post = r"[\"\']\)\s*\}\})"
+rgx_sel = r"(?P<selector>\s*\#\s+\[[\w\s()<>!=.,\-\'\"]+\])?"
+
+pat_compiler_c = re.compile("".join([rgx_idt, rgx_pre, "c", rgx_post, rgx_sel]))
+pat_compiler_m2c = re.compile("".join([rgx_idt, rgx_pre, "m2w64_c", rgx_post, rgx_sel]))
+pat_compiler_other = re.compile(
+    "".join([rgx_idt, rgx_pre, "(m2w64_)?(cxx|fortran)", rgx_post, rgx_sel])
+)
+pat_compiler = re.compile(
+    "".join([rgx_idt, rgx_pre, "(m2w64_)?(c|cxx|fortran)", rgx_post, rgx_sel])
+)
+pat_stdlib = re.compile(r".*\{\{\s*stdlib\([\"\']c[\"\']\)\s*\}\}.*")
+# no version other than 2.17 currently available (except 2.12 as default on linux-64)
+pat_sysroot_217 = re.compile(r"- sysroot_linux-64\s*=?=?2\.17")
```

Comment:
```suggestion
pat_sysroot = re.compile(r"- sysroot_linux-64\s*=?=?2\.(17|28)")
```

We have 2.28 in the wild right now and formally people can build with it if they do not use cdts. Maybe we should account for it now?

Reply:
I had checked at the time and there were no users of 2.28. Now that I've expanded the [search](https://github.com/search?q=org%3Aconda-forge+path%3Arecipe%2Fmeta.yaml+%2F-+sysroot_%28linux-64%7C%5C%7B%5C%7B.*%5C%7D%5C%7D%29%5Cs*%3D%3F%3D%3F2%5C.28%2F&type=code) a bit, there's only cudnn (which seems misspecified actually because it pins both 2.17 & 2.28 on aarch).

Reply:
Yep. As I meant to say totally optional.